python_modules/dagster-test/dagster_test/toys/partitioned_assets/partitioned_run_request_sensors.py

```diff
@@ -0,0 +1,49 @@
+import random
+
+from dagster import AssetSelection, RunRequest, SensorResult, define_asset_job, sensor
+
+from .dynamic_asset_partitions import ints_dynamic_asset, ints_dynamic_partitions_def
+from .hourly_and_daily_and_unpartitioned import (
+    upstream_daily_partitioned_asset,
+)
+
+
+@sensor(asset_selection=AssetSelection.assets(ints_dynamic_asset))
+def ints_asset_selection_sensor(context):
+    new_partition_key = str(random.randint(0, 100))
+    return SensorResult(
+        run_requests=[RunRequest(partition_key=new_partition_key)],
+        dynamic_partitions_requests=[
+            ints_dynamic_partitions_def.build_add_request([new_partition_key])
+        ],
+    )
+
+
+@sensor(
+    job=define_asset_job(
+        "ints_job",
+        AssetSelection.assets(ints_dynamic_asset),
+        partitions_def=ints_dynamic_partitions_def,
+    )
+)
+def ints_job_sensor():
```

Comment:
Can we give this sensor (and the job) a name that indicates what its purpose is? dynamic_partitions_job_sensor maybe?

Reply:
good call